### PR TITLE
Add workshop case study

### PR DESCRIPTION
**Pull request by tarnold17**
## Fixes/Addresses/Summary/Motivation:

(Replace this with the issue # fixed or addressed in this PR as well as any other related PRs.  If
no issue or PRs exist, then add a brief statement on what changes this PR is bringing in.)

## Changes proposed in this PR:
- Add input file for PARETO workshop case study
- Add tests for workshop case study
- Add documentation for workshop case study

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my
contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: docs/case_studies/index.rst [hunk 1]
```diff
@@ -86,73 +98,85 @@ Comparison table
...
      - No
      - Yes, for K03 and K05
+     - Yes
      - No
    * - **Storage sites**
      - 3
      - 1
      - 2
      - 3
+     - 3
```
Comment (melody-shellman) [inline docs/case_studies/index.rst:164]: I believe there is only one storage site actually connected to the network (S02) - let me know if I'm missing something. S03 and S04 are defined in the list of storage sites but are not connected to any other node in the system and are not pictured in the diagram.
```suggestion
     - 1
```

File: pareto/tests/test_strategic_model.py [hunk 2]
```diff
@@ -1598,3 +1592,173 @@ def test_infrastructure_buildout(build_toy_strategic_model):
...
+        "running_time": 60 * 5,
+        "gap": 0,
+    }
+
+    results = solve_model(model=m, options=options)
+
+    assert results.solver.termination_condition == pyo.TerminationCondition.optimal
+    assert results.solver.status == pyo.SolverStatus.ok
+    assert degrees_of_freedom(m) == 3851
+    assert pytest.approx(5661.39656, abs=1e-1) == pyo.value(m.v_Z)
```
Comment (melody-shellman) [inline pareto/tests/test_strategic_model.py:1762]: Nice to have a case study that can be solved to optimality with CBC in short time.

**Review by melody-shellman — COMMENTED** [review]
One item in table documentation that needs addressing. Should be quick.

File: docs/case_studies/index.rst (hunk 1 above)
Comment (tarnold17) [reply docs/case_studies/index.rst:164]: @melody-shellman S03 and S04 are not used in the workshop baseline scenario, but are included in the input file so they can be used in the beneficial reuse scenario which is based off of the baseline scenario. I added a clarifying note to the documentation.

**Review by melody-shellman — APPROVED** [review]
Looks good!

File: docs/case_studies/index.rst (hunk 1 above)
Comment (melody-shellman) [reply docs/case_studies/index.rst]: okay! that's clear. Looks good.

**Review by ElmiraShamlou — APPROVED** [review]
LGTM

**Pull request merged**